rest/config.go

```diff
@@ -809,6 +809,21 @@ func (config *StartupConfig) Serve(addr string, handler http.Handler) error {
 	)
 }
 
+// Validate insecure connections
+func (sc *StartupConfig) validateInsecureTLSConnection() (err error) {
+	// Validate SSL is provided if not allowing unsecure connections
+	if sc.API.HTTPS.AllowInsecureTLSConnections == nil || !*sc.API.HTTPS.AllowInsecureTLSConnections {
+		if sc.API.HTTPS.TLSKeyPath == "" || sc.API.HTTPS.TLSCertPath == "" {
+			return fmt.Errorf("a TLS key and cert path must be provided when not allowing insecure TLS connections")
```

Comment:
The double negative in this message is a bit confusing. Think we can rephrase this to be:

```suggestion
			return fmt.Errorf("TLS key path and cert path must be provided unless api.https.allow_insecure_tls_connections is set")
```